src/xenia/gpu/graphics_system.cc

```diff
@@ -118,6 +118,20 @@ X_STATUS GraphicsSystem::Setup(cpu::Processor* processor,
       reinterpret_cast<cpu::MMIOReadCallback>(ReadRegisterThunk),
       reinterpret_cast<cpu::MMIOWriteCallback>(WriteRegisterThunk));
 
+  // GPU counters update thread
+  auto counters_worker_thread = kernel::object_ref<kernel::XHostThread>(
+      new kernel::XHostThread(kernel_state_, 128 * 1024, 0, [this]() {
+        while (1) {
+          WriteRegister(
```

Comment:
This looks like a race condition between the new thread and the command processor, though not sure how it's handled with MMIO writes currently.